src/notebooks/execution/cellExecution.ts

```diff
@@ -495,7 +511,7 @@ export class CellExecution implements IDisposable {
             // Compute the hash for the cell we're about to execute if on the interactive window
             let hash: ICellHash | undefined = undefined;
             if (this.cell.notebook.notebookType === InteractiveWindowView) {
-                hash = await this.cellHashProvider.addCellHash(this.cell);
+                hash = await this.generatedCodeStore?.getCellHash(this.cell);
```

Comment:
Could you skip checking for interactive window? Hash would be empty for notebooks and then this code doesn't need to know about interactive.

Reply:
Agreed, thats possible. Still don't like the fact that we have to inject this thing, yuck..
Anways

Reply:
I thought you figured out a way to have it just show up in the metadata. No injecting.

If I recall that's why we needed the willExecute hook.

await willExecute hook
get interactive metadata

Reply:
Can do, but the concern @IanMatthewHuff  had was as follows:
* In one place we get the information out to the metadata
* In the other place we use the storage provider.
I'm torn over this, I would love to remove the storage provider, but agree with Ian, there's a descrepency, and we should stick to just one way to get the stuff, not two. I.e. not store the data in metadata as well as dictionary.

FYI - we need a dictionary as users can always clear all of the cells in the IW & start debugging a cell (at which point all of the cell hash information is lost - as its stored in the cells).

Reply:
Happy to change this, but as mentioned torn between the two approachs (one less dependency vs two places to store the same information)

Reply:
@rchiodo made the changes, I think it like it better now.